### PR TITLE
[FAB-17591] Support adding an orderer endpoint to existing config

### PR DESCRIPTION
**Pull request by wuxuer**
Description

Modify an existing channel configuration to add an orderer endpoint to an orderer org's config group.

Related issues

https://jira.hyperledger.org/browse/FAB-17591

Signed-off-by: xu wu <wuxu1103@163.com>

#### Type of change

- New feature

### CODE REVIEW

File: pkg/config/orderer.go [hunk 1]
```diff
@@ -284,3 +284,40 @@ func addOrdererPolicies(cg *cb.ConfigGroup, policyMap map[string]Policy, modPoli
...
+	for _, e := range existingOrdererEndpoints {
+		if e == endpoint {
+			return fmt.Errorf("orderer org %s already contains endpoint %s",
+				orgName, endpoint)
+		}
+	}
+
+	existingOrdererEndpoints = append(existingOrdererEndpoints, endpoint)
+
+	// Add anchor peers config value back to application org
```
Comment (caod123) [inline pkg/config/orderer.go]: `Add orderer endpoints config value back to orderer org`

**Review by caod123 — CHANGES_REQUESTED** [review]

**Comment by caod123** [comment]
@hellosmallstone can you also update the example_test update config scenario to reflect adding an orderer endpoint? Thanks!

File: pkg/config/orderer.go (hunk 1 above)
Comment (wuxuer) [reply pkg/config/orderer.go]: Yeah, I will fix it.

**Comment by wuxuer** [comment]
> @hellosmallstone can you also update the example_test update config scenario to reflect adding an orderer endpoint? Thanks!

OK, I will update it.

**Comment by wuxuer** [comment]
@caod123 Hi, I have updated the example_test. Please review it, thx.
For some unknown reason, I can not send you a "Re-request review" info via the web button. Therefore, I add this comment.

**Comment by caod123** [comment]
LGTM +1

File: pkg/config/orderer.go [hunk 2]
```diff
@@ -284,3 +284,40 @@ func addOrdererPolicies(cg *cb.ConfigGroup, policyMap map[string]Policy, modPoli
 
 	return addPolicies(cg, policyMap, modPolicy)
 }
+
+// AddOrdererEndpoints adds an orderer's endpoint to an existing channel config transaction.
```
Comment (sykesm) [inline pkg/config/orderer.go:288]: `AddOrdererEndpoint` vs `AddOrdererEndpoints` - the Doc name doesn't match the function name.

File: pkg/config/orderer.go [hunk 3]
```diff
@@ -284,3 +284,40 @@ func addOrdererPolicies(cg *cb.ConfigGroup, policyMap map[string]Policy, modPoli
...
+	if !ok {
+		return fmt.Errorf("orderer org %s does not exist in channel config", orgName)
+	}
+
+	ordererAddrProto := &cb.OrdererAddresses{}
+
+	if ordererAddrConfigValue, ok := ordererOrgGroup.Values[EndpointsKey]; ok {
+		err := proto.Unmarshal(ordererAddrConfigValue.Value, ordererAddrProto)
+		if err != nil {
+			return fmt.Errorf("failed unmarshalling orderer org %s's  endpoints: %v", orgName, err)
```
Comment (sykesm) [inline pkg/config/orderer.go:302]: two spaces in the message instead of one and we can omit the possessive.

**Review by sykesm — APPROVED** [review]
nits can be fixed on top.

@caod123 @stephyee

File: pkg/config/orderer.go (hunk 2 above)
Comment (wuxuer) [reply pkg/config/orderer.go:288]: Sorry about it. 
I fixed the two nits in https://github.com/hyperledger/fabric/pull/891.

**Pull request merged**